File: backend/app/schemas/analyze.py
from pydantic import BaseModel, Field


class Suggestion(BaseModel):
    title: str
    detail: str


class AnalyzeResponse(BaseModel):
    match_score: int = Field(..., ge=0, le=100)
    summary: str
    matches: list[str]
    gaps: list[str]
    suggestions: list[Suggestion]
    cached: bool = False


class StatsResponse(BaseModel):
    total_analyses: int
    average_score: float

File: frontend/src/types/analyze.ts
export interface Suggestion {
  title: string
  detail: string
}

export interface AnalyzeResponse {
  match_score: number
  summary: string
  matches: string[]
  gaps: string[]
  suggestions: Suggestion[]
  cached: boolean
}

export interface StatsResponse {
  total_analyses: number
  average_score: number
}
